audio-device-client/idl-example.md

```diff
@@ -0,0 +1,113 @@
+**Note: that this content is work-in-progress at the early stage of
+development.**
+
+
+# WebIDL (WIP)
+
+### AudioDeviceClientState
+An `enum` represents the current state of the client.
+
+```webidl
+enum AudioDeviceClientState {
+  "pending",
+  "running",
+  "stopped"
+}
+```
+
+### AudioDeviceClient
+
+The low-level representation of an audio device. Provides various
+hardware-related properties and a path to the associated `AudioContext` for Web
+Audio API tasks.
+
+```webidl
+[Exposed=Window, SecureContext, NoConstructor]
+interface AudioDeviceClient : EventTarget {
+  readonly attribute USVString deviceId;
+  readonly attribute float sampleRate;
+  readonly attribute unsigned callbackBufferSize;
+  readonly attribute unsigned inputChannelCount;
+  readonly attribute unsigned outputChannelCount;
+  readonly attribute AudioDeviceClientState state;
+  readonly attribute MessagePort port;
+  attribute EventHandler onstatechange;
+  Promise<void> start(USVString moduleUrl);
+  Promise<void> stop();
+  [SameObject] AudioContext getContext();
+}
```

Comment:
Let's not make the same mistake as WebAudio.  you **MUST** close the client if you want it to go away.  (Or is that what `stop()` does?)

Reply:
Yes. `stop()` is supposed to do that. IDL is tentative, so we can continue the discussion once this gets published.